src/shared/utils/hooks/useDebounce.ts

```diff
@@ -0,0 +1,24 @@
+import { useEffect, useState } from 'react'
+
+export function useDebounce(value: string, delay: number) {
+  // value and delay in ms (1000ms = 1s)
+  // debounced values
+  const [debouncedValue, setDebouncedValue] = useState(value)
+
+  useEffect(
+    () => {
+      // Update debounced value after delay
```

Comment:
if (delay < 0) {
      console.warn('Delay should not be negative')
      return
    }